File: SchedRPG/app/src/main/java/com/vladv/questsched/tabs/TaskCreationTab.java
package com.vladv.questsched.tabs;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ArrayAdapter;

import androidx.appcompat.app.AppCompatActivity;

import com.example.schedrpg.databinding.ActivityTaskCreationTabBinding;
import com.vladv.questsched.user.UserTask;
import com.vladv.questsched.myfirebasetool.ChangeFirebaseData;
import com.vladv.questsched.user.User;

public class TaskCreationTab extends AppCompatActivity {

    ActivityTaskCreationTabBinding binding;
    User user = new User();


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityTaskCreationTabBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);

        String[] stats = new String[]{"Strength" ,
                "Dexterity" ,
                "Constitution" ,
                "Wisdom" ,
                "Intelligence" ,
                "Charisma"};

        ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.simple_spinner_dropdown_item, stats);
        binding.createTaskType.setAdapter(adapter);

        String[] difficulty = new String[]{"very easy", "easy", "medium", "hard", "very hard"};

        ArrayAdapter<String> adapter2 = new ArrayAdapter<>(this, android.R.layout.simple_spinner_dropdown_item, difficulty);
        binding.createTaskDifficulty.setAdapter(adapter2);

        binding.createTaskSubmit.setOnClickListener(v -> addTaskToFirebase());

        binding.createTaskBack.setOnClickListener(v -> startActivity(new Intent(TaskCreationTab.this, LogInSuccess.class)));

    }

    private void addTaskToFirebase()
    {
        String name = binding.createTaskName.getText().toString().trim();
        String description = binding.createTaskDescription.getText().toString().trim();
        int difficulty = binding.createTaskDifficulty.getSelectedItemPosition();
        int type = binding.createTaskType.getSelectedItemPosition();

        if(name.isEmpty())
        {
            binding.createTaskName.setError("Name is required.");
            binding.createTaskName.requestFocus();
            return;
        }

        UserTask userTask = new UserTask(type,difficulty,name,description);
        user.addTask(userTask);
        System.out.println(user.toString());

        ChangeFirebaseData changeFirebaseData = new ChangeFirebaseData();
        changeFirebaseData.updateUserData(user);




    }
}

File: SchedRPG/app/src/main/java/com/vladv/questsched/tabs/TaskCreationTab.kt
package com.vladv.questsched.tabs

import android.R
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.ArrayAdapter
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.schedrpg.databinding.ActivityTaskCreationTabBinding
import com.vladv.questsched.myfirebasetool.ChangeFirebaseData
import com.vladv.questsched.user.User
import com.vladv.questsched.user.UserTask

class TaskCreationTab : AppCompatActivity() {
    private var binding: ActivityTaskCreationTabBinding? = null
    private var user = User()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityTaskCreationTabBinding.inflate(
            layoutInflater
        )
        val view: View = binding!!.root
        setContentView(view)
        val stats = arrayOf(
            "Strength",
            "Dexterity",
            "Constitution",
            "Wisdom",
            "Intelligence",
            "Charisma"
        )
        val adapter = ArrayAdapter(this, R.layout.simple_spinner_dropdown_item, stats)
        binding!!.createTaskType.adapter = adapter
        val difficulty = arrayOf("very easy", "easy", "medium", "hard", "very hard")
        val adapter2 = ArrayAdapter(this, R.layout.simple_spinner_dropdown_item, difficulty)
        binding!!.createTaskDifficulty.adapter = adapter2
        binding!!.createTaskSubmit.setOnClickListener { addTaskToFirebase() }
        binding!!.createTaskBack.setOnClickListener {
            startActivity(
                Intent(
                    this@TaskCreationTab,
                    LogInSuccess::class.java
                )
            )
        }
    }

    private fun addTaskToFirebase() {
        val name = binding!!.createTaskName.text.toString().trim { it <= ' ' }
        val description = binding!!.createTaskDescription.text.toString().trim { it <= ' ' }
        val difficulty = binding!!.createTaskDifficulty.selectedItemPosition
        val type = binding!!.createTaskType.selectedItemPosition
        if (name.isEmpty()) {
            binding!!.createTaskName.error = "Name is required."
            binding!!.createTaskName.requestFocus()
            return
        }
        val userTask = UserTask(type, difficulty, name, description)
        user.addTask(userTask)
        val changeFirebaseData = ChangeFirebaseData()
        changeFirebaseData.updateUserData(user)
        makeToast("Task: " + userTask.name + " created succesfully")
        val intent = intent
        finish()
        startActivity(intent)
    }

    private fun makeToast(s: String) {
        Toast.makeText(TasksViewTab.context, s, Toast.LENGTH_SHORT).show()
    }
}